test/Common/TestHelpers.cs

```diff
@@ -257,6 +256,26 @@ public static DurableTaskOptions GetDurableTaskOptionsForStorageProvider(string
             }
         }
 
+        /// <summary>
+        /// Read a file's contents, line by line, even if another process is currently writing to it.
```

Comment:
Otherwise we get an exception about not being able to read the log-file's contents because another process is using it

Reply:
It's a tricky one :) . Before, I was using `File.ReadAllLines` which apparently attempts to open the file using a restrictive, read-only, file permission. This is incompatible with our file logger's write permissions, so it errors out. With this, we are opening the file using a "read-write" permission so it works just fine :)

Reply:
I got the description, **and the method itself,** from this SO answer: https://stackoverflow.com/questions/12744725/how-do-i-perform-file-readalllines-on-a-file-that-is-also-open-in-excel

Reply:
Let's directly link to that SO answer in our comment.

Reply:
I didn't see this before, I'll link it